src/main/java/com/aws/greengrass/deployment/KernelUpdateDeploymentTask.java

```diff
@@ -49,6 +53,9 @@ public class KernelUpdateDeploymentTask implements DeploymentTask {
     public KernelUpdateDeploymentTask(Kernel kernel, Logger logger, Deployment deployment,
                                       ComponentManager componentManager) {
         this.kernel = kernel;
+        this.kernelAlternatives = kernel.getContext().get(KernelAlternatives.class);
+        this.deploymentDirectoryManager = kernel.getContext().get(DeploymentDirectoryManager.class);
+        this.bootstrapManager = kernel.getContext().get(BootstrapManager.class);
```

Comment:
these are only used in `call` within an `if`, so pull them from the context when they're actually needed, not in the constructor.

Reply:
I initially was pulling this from context when required, however, I do not want to pull, say `KernelAlternatives`, from context in the same function twice.

Reply:
So save it in a local variable within the method then.

Reply:
updated